aws/resource_aws_cloudwatch_metric_alarm_test.go

```diff
@@ -236,6 +236,26 @@ func TestAccAWSCloudWatchMetricAlarm_extendedStatistic(t *testing.T) {
 	})
 }
 
+func TestAccAWSCloudWatchMetricAlarm_expression(t *testing.T) {
+	var alarm cloudwatch.MetricAlarm
+	rInt := acctest.RandInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloudWatchMetricAlarmDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudWatchMetricAlarmConfigWithExpression(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudWatchMetricAlarmExists("aws_cloudwatch_metric_alarm.foobar", &alarm),
+					resource.TestCheckResourceAttr("aws_cloudwatch_metric_alarm.foobar", "metric_query.#", "2"),
+				),
+			},
```

Comment:
Since the usage of expressions does not require recreating the Terraform resource, we should include another `TestStep` similar to the first which provides an updated test configuration (e.g. adding another `metric_query` configuration and including it in the expression), e.g.

```go
				Config: testAccAWSCloudWatchMetricAlarmConfigWithExpressionUpdated(rInt),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckCloudWatchMetricAlarmExists("aws_cloudwatch_metric_alarm.foobar", &alarm),
					resource.TestCheckResourceAttr("aws_cloudwatch_metric_alarm.foobar", "metric_query.#", "3"),
				),
```

As well as a `TestStep` that verifies these new configurations can be properly imported as the `aws_cloudwatch_metric_alarm` resource currently supports resource import, e.g.

```go
{
  ResourceName: "aws_cloudwatch_metric_alarm",
  ImportState: true,
  ImportStateVerify: true,
},
```